pkg/v1/providers/infrastructure-vsphere/v1.1.0/ytt/overlay.yaml

```diff
@@ -316,6 +311,15 @@ spec:
     #@overlay/append
     - sed -i '/listen-client-urls/ s/$/,https:\/\/127.0.0.1:2379/' /etc/kubernetes/manifests/etcd.yaml
     #@ end
+    #! When using kube-vip as the control plane endpoint provider with IPv6 as
+    #! the primary IP family, set --node-ip on kubelet so that host network pods
+    #! do not get the kube-vip address as their pod IP.
+    #! See: https://github.com/vmware-tanzu/tanzu-framework/issues/2098
+    #@ if not data.values.AVI_CONTROL_PLANE_HA_PROVIDER and data.values.TKG_IP_FAMILY in ["ipv6", "ipv6,ipv4"]:
```

Comment:
Is it safe to add this overlay when the LB is neither AVI nor kube-vip? I'm guessing there's no way to determine from data values that kube-vip has been selected, hence the !AVI?

Reply:
Good question. I think the default (blank) is kube-vip and there aren't currently any other control plane endpoint providers I'm aware of. I saw some prior art checking things this way, but perhaps there's a more precise way to check whether we're using kube-vip.

@lubronzhan do you have any input on this?

Reply:
hm, i don't think there is precise way to do it at the moment unfortunately.

Reply:
OK, marking this resolved since there doesn't seem to be an obviously better answer at this time and this matches the existing pattern.